src/main/java/at/hannibal2/skyhanni/test/TestCopyChestData.kt

```diff
@@ -0,0 +1,82 @@
+package at.hannibal2.skyhanni.test
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.test.command.CopyItemCommand.grabItemData
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.KeyboardManager.isKeyHeld
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.OSUtils
+import net.minecraft.client.Minecraft
+import net.minecraft.client.gui.inventory.GuiChest
+import net.minecraft.client.gui.inventory.GuiContainer
+import net.minecraft.inventory.Slot
+import net.minecraft.item.ItemStack
+import net.minecraftforge.client.event.GuiScreenEvent
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+object TestCopyChestData {
+    private val config get() = SkyHanniMod.feature.dev.debug.copyInventory
+    private val mc = Minecraft.getMinecraft()
+    private const val HOTBAR_SIZE = 9
+    private const val INVENTORY_SIZE = 27
+    private const val DEBUG_SUFFIX: String = "data copied into the clipboard! §lMake sure to save it into a .txt file; these tend to get quite long."
+    @SubscribeEvent
+    fun onKeybind(event: GuiScreenEvent.KeyboardInputEvent.Post) {
+        if (!config.copyPlayerInventory.isKeyHeld() && !config.copyEntireChest.isKeyHeld() && !config.copyChestName.isKeyHeld()) return
+        if (event.gui !is GuiContainer) return
+        if (config.copyEntireChest.isKeyHeld() && event.gui is GuiChest) {
+            copyChestData(chest = (event.gui as GuiChest).inventorySlots.inventorySlots)
+            return
+        } else if (config.copyPlayerInventory.isKeyHeld()) {
+            copyInventoryData(inventory = if (config.includeArmor) mc.thePlayer.inventory.mainInventory + mc.thePlayer.inventory.armorInventory else mc.thePlayer.inventory.mainInventory)
+            return
+        } else if (InventoryUtils.openInventoryName().isNotEmpty()) {
+            OSUtils.copyToClipboard(InventoryUtils.openInventoryName())
+            ChatUtils.chat("Chest name copied to clipboard.")
+            return
+        }
+    }
+    private fun copyInventoryData(inventory: Array<ItemStack?>) {
```

Comment:
this function is so overcomplicated. My idea split the array into 3 sub list. Make a sepreate function that handles the item printing. Also make it contsistent. Eg. Slot Number : Item Info. This is mainly the issue with the null display.